src/core/Akka/Actor/Futures.cs

```diff
@@ -178,14 +178,14 @@ public static Task<T> Ask<T>(this ICanTell self, Func<IActorRef, object> message
         /// <returns>Provider used for Ask pattern implementation</returns>
         internal static IActorRefProvider ResolveProvider(ICanTell self)
         {
-            if (self is ActorSelection)
-                return ResolveProvider(self.AsInstanceOf<ActorSelection>().Anchor);
```

Comment:
BTW, I noticed this while doing my work on Akka.NET v1.4.25 - `AsInstanceOf` boxes / unboxes each time it's used. Getting rid of it or making it generic wouldn't be a bad way to improve performance across the board. It's used like 600 times - much of that is the test suite, but a lot of it is in performance-critical pathways too.

cc @to11mtm

Reply:
Wouldn't be a bad idea to make it generic, and/or mark it with a `where T:class` so that anything with a struct will blow up, which would be the spots where I'd worry about boxing/unboxing.

Reply:
Alternative idea - use a Roslyn analyzer + fixup to replace the `AsInstanceOf` classes with something more appropriate.

We can also turn on nullability for Akka.NET v1.5 and tack this issue onto that milestone.

Reply:
Just for clarity, I don't think a Nullable check itself will help. The issue in `.AsInstanceOf` is that there is no way to be sure that the cast itself is valid, not whether the object is or is not null.

Refer to my gitter comments RE: Roslyn analyzer. :)